production-audit/server-snapshot/retry-proxy/gpt56_retry_proxy.py

```
@@ -0,0 +1,281 @@
+                            self.wfile.flush()
+                        return
+            except urllib.error.HTTPError as err:
+                response_body = err.read()
+                if err.code in RETRY_HTTP_CODES and attempt < ProxyConfig.attempts:
+                    time.sleep(ProxyConfig.backoff * attempt)
+                    continue
+                self.send_upstream_response(err.code, dict(err.headers.items()), response_body)
+                return
+            except (BrokenPipeError, ConnectionResetError):
+                return
+            except Exception as exc:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)